runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -369,6 +362,51 @@ public void outputWithTimestamp(OutputT output, Instant timestamp) {
       checkNotNull(combiner, "Combiner passed to createAggregator cannot be null");
       return aggregatorFactory.createAggregatorForDoFn(fn.getClass(), stepContext, name, combiner);
     }
+
+    @Override
+    public BoundedWindow window() {
+      throw new UnsupportedOperationException("Cannot access window.");
+    }
+
+    @Override
+    public Context context(DoFn<InputT, OutputT> doFn) {
+      return this;
+    }
+
+    @Override
+    public ProcessContext processContext(DoFn<InputT, OutputT> doFn) {
+      throw new UnsupportedOperationException("Cannot access ProcessContext.");
```

Comment:
In a closed-world scenario, each of these exceptions would be an `IllegalStateException` as it is the requirement that `DoFnSignatures` makes it impossible to call. But I am trying to stick with context unaware programming here, especially considering the rate of change of this area of the code.